engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/ConfirmUrlPopup.java

```diff
@@ -73,7 +76,23 @@ public void setMessage(String title, String message) {
         }
     }
 
-    public void setCheckbox(WebBrowserConfig webBrowserConfig, String url) {
+    public void setCheckbox(WebBrowserConfig webBrowserConfig, String url) throws MalformedURLException {
+    	UICheckbox saveHostName = find("saveHostName", UICheckbox.class);
+
+        if (saveHostName != null && webBrowserConfig != null) {
+            saveHostName.setChecked(false);
+            String hostName = new URL(url).getHost();
+
+            saveHostName.subscribe(checkbox -> {
+                boolean isTrustedHostName = saveHostName.isChecked();
+
+                if (isTrustedHostName) {
+                    webBrowserConfig.addTrustedHostName(hostName);
+                } else {
+                    webBrowserConfig.removeTrustedHostName(hostName);
+                }
+            });
+            
         UICheckbox saveUrl = find("saveUrl", UICheckbox.class);
 
         if (saveUrl != null && webBrowserConfig != null) {
```

Comment:
I don't think this is right. You've got this if statement inside the prior on that checks for the `saveHostName`.  

They should either both be seperate or they should combined into one statement.  
I would recommend seperate.

Reply:
This was not my implementation. This previously existed in the repository.

Reply:
Also, Java ignores white spaces. It's just a styling tweak.